ledger-core/virtual/integration/method_test.go

```diff
@@ -978,3 +982,174 @@ func TestVirtual_CallMethodFromConstructor_Unordered(t *testing.T) {
+				objectState       = &payload.VStateReport_ProvidedContentBody{
+					LatestValidatedState: &payload.ObjectState{
+						Reference: validatedStateRef,
+						Class:     class,
+						State:     []byte("12345"),
+					},
+					LatestDirtyState: &payload.ObjectState{
+						Reference: dirtyStateRef,
+						Class:     class,
+						State:     []byte("54321"),
+					},
+				}
+			)
+
+			{ // send object state to server
+				pl := payload.VStateReport{
+					Status:                        payload.Ready,
+					Object:                        objectGlobal,
+					LatestValidatedState:          validatedState,
+					LatestDirtyState:              dirtyState,
+					ProvidedContent:               objectState,
+					UnorderedPendingEarliestPulse: pulse.OfNow(),
+				}
+
+				server.WaitIdleConveyor()
+				server.SendPayload(ctx, &pl)
+				server.WaitActiveThenIdleConveyor()
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+			{
+				typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+					require.Equal(t, res.ReturnArguments, []byte("345"))
+					require.Equal(t, res.Callee, objectGlobal)
+
+					return false // no resend msg
+				})
+
+				pl := payload.VCallRequest{
+					CallType:            payload.CTMethod,
+					CallFlags:           payload.BuildCallFlags(contract.CallIntolerable, test.state),
+					CallAsOf:            0,
+					Caller:              server.GlobalCaller(),
+					Callee:              objectGlobal,
+					CallSiteDeclaration: class,
+					CallSiteMethod:      "Test",
+					CallOutgoing:        objectLocal,
+					Arguments:           insolar.MustSerialize([]interface{}{}),
+				}
+
+				key := calculateOutgoing(pl).String()
+				runnerMock.AddExecutionMock(key).
+					AddStart(func(ctx execution.Context) {
+						require.Equal(t, objectGlobal, ctx.Object)
+						require.Equal(t, test.state, ctx.Request.CallFlags.GetState())
+						require.Equal(t, test.state, ctx.Isolation.State)
+						require.Equal(t, objectGlobal, ctx.ObjectDescriptor.HeadRef())
+						stateClass, err := ctx.ObjectDescriptor.Class()
+						require.NoError(t, err)
+						require.Equal(t, class, stateClass)
+
+						for _, checkFn := range test.checks {
+							require.True(t, checkFn(objectState, ctx.ObjectDescriptor))
+						}
+					}, &executionupdate.ContractExecutionStateUpdate{
+						Type:   executionupdate.Done,
+						Result: requestresult.New([]byte("345"), objectGlobal),
+					})
+				runnerMock.AddExecutionClassify(key, contract.MethodIsolation{
+					Interference: contract.CallIntolerable,
+					State:        test.state,
+				}, nil)
+
+				server.SendPayload(ctx, &pl)
+			}
+
+			if timeout := waitServerDone(executeDone, server); timeout != nil {
+				require.FailNow(t, timeout.Error())
+			}
+
+			assert.Equal(t, 1, typedChecker.VCallResult.Count())
+
+			mc.Finish()
+		})
+	}
+}
+
+func waitServerDone(execDone synckit.SignalChannel, server *utils.Server) error {
```

Comment:
May you can use this utility https://github.com/insolar/assured-ledger/pull/369

Reply:
done